platforms/jvm/language-jvm/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java

```diff
@@ -482,6 +484,17 @@ public Property<Integer> getRelease() {
         return release;
     }
 
+    /**
+     * Configures whether Preview Features should be enabled for the compilation ({@code --enable-preview} compiler flag).
+     *
+     * @since 8.8
+     */
+    @Incubating
+    @Input
+    @Optional
+    public Property<Boolean> getEnablePreview() {
+        return enablePreview;
+    }
```

Comment:
:x: This should be declared as an abstract property. I don't think you even need to mention it in the implementation class, just let the interface declare it and you should be good.

Reply:
`CompileOptions` is a public API, and adding an abstract method would be backward-incompatible. I wouldn't want to wait for a Gradle major with this change.

I don't know if it is supported to use a throwing stub implementation like we do with injected services. But then it does not reduce the complexity much, and I would stick with a regular field, like all the other values in this class use.

Reply:
I can bow to that.